### PR TITLE
NODE-484 Mass Transfer Transaction

### PR DESCRIPTION
**Pull request by petermz**

### CODE REVIEW

File: src/main/scala/scorex/transaction/assets/MassTransferTransaction.scala [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+             recipients: List[(AddressOrAlias, Long)],
+             timestamp: Long,
+             feeAmount: Long,
+             attachment: Array[Byte]): Either[ValidationError, MassTransferTransaction] = {
+    create(assetId, sender, recipients, timestamp, feeAmount, attachment, ByteStr.empty).right.map { unsigned =>
+      unsigned.copy(signature = ByteStr(EllipticCurveImpl.sign(sender, unsigned.toSign())))
+    }
+  }
+
+  def processRecipientsWith[T, R](recipients: List[(T, Long)])(f: (T, Long) => Validation[R]): Validation[List[R]] =
```
Comment (ismagin) [inline src/main/scala/scorex/transaction/assets/MassTransferTransaction.scala]: What is `T`? `AddressOrAlias` ?
Isn't it just `.traverse` ?


`def traverse[G[_]: Applicative, B](f : C => G[B]): G[F[B]]`

File: src/test/scala/com/wavesplatform/TransactionGen.scala [hunk 2]
```diff
@@ -178,6 +182,19 @@ trait TransactionGen {
     (assetId, sender, _, amount, timestamp, feeAssetId, feeAmount, attachment) <- transferParamGen
   } yield TransferTransaction.create(assetId, sender, sender, amount, timestamp, feeAssetId, feeAmount, attachment).right.get
 
+  val massTransferGen = (
+      for {
+        (assetId, sender, _, _, timestamp, _, feeAmount, attachment) <- transferParamGen
+        maxTransfers = MassTransferTransaction.MaxTransferCount
```
Comment (ismagin) [inline src/test/scala/com/wavesplatform/TransactionGen.scala]: Make this Gen.choose(0, MassTransferTransaction.MaxTransferCount)?
Can it be zero? Looks like it can now. If this should be disabled, consider adding validation or use `NonEmptyList[T]`

File: src/test/scala/com/wavesplatform/state2/diffs/MassTransferTransactionDiffTest.scala [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+              }
+            }
+          }
+        }
+      }
+    }
+
+    import MassTransferTransaction.{MaxTransferCount => Max}
+    Seq(0, 1, Max) foreach testDiff // test edge cases
+    Gen.choose(2, Max - 1) map testDiff
```
Comment (ismagin) [inline src/test/scala/com/wavesplatform/state2/diffs/MassTransferTransactionDiffTest.scala]: What does this line do?

File: src/test/scala/com/wavesplatform/state2/diffs/MassTransferTransactionDiffTest.scala [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  val fs = TestFunctionalitySettings.Enabled.copy(preActivatedFeatures = Map(BlockchainFeatures.MassTransfer.id -> 0))
+
+  val baseSetup: Gen[(GenesisTransaction, PrivateKeyAccount, Long)] = for {
+    master <- accountGen
+    ts <- positiveLongGen
+    genesis: GenesisTransaction = GenesisTransaction.create(master, ENOUGH_AMT, ts).right.get
+  } yield (genesis, master, ts)
+
+  property("MassTransfer preserves balance invariant") {
+    def testDiff(transferCount: Int) = {
```
Comment (ismagin) [inline src/test/scala/com/wavesplatform/state2/diffs/MassTransferTransactionDiffTest.scala]: Why not make `transferCount` generator-based too? `Gen.choose` would do
